### PR TITLE
Updating messaging doc for changes to editing, deleting, and linking features

### PR DESCRIPTION
**Pull request by JeffSchering**
In EE, System Admins can configure if and when users can edit sent messages, and who can delete sent messages.

Also, the timestamp on each message is a permalink.

### CODE REVIEW

File: source/help/messaging/sending-messages.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+-----------------
+
+Write a message by typing into the text input box, then press **ENTER** to send it. Use **SHIFT+ENTER** to create a new line without sending a message. To send messages by pressing **CTRL+ENTER**, go to **Main Menu > Account Settings > Send messages on Ctrl + Enter**.
+
+Editing a Message
+-----------------
+
+The ability to edit a message after it is sent depends on which edition you are using.
+
+Team Edition
```
Comment (esethna) [inline source/help/messaging/sending-messages.rst]: End users won't understand the term "Enterprise edition". Propose we can just say:

"Edit a message by clicking the **[...]** icon next to any message that you’ve sent, then click **Edit**. When you're done editing, press **ENTER** to save the changes. Message editing may be time limited or disabled by your System Admin.

Note: Edited messages do not trigger new mention notifications, desktop notifications, or notification sounds. "

File: source/help/messaging/sending-messages.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  System Admins can turn editing off, or they can set a time limit. If a time limit is set, you can edit a message only if the time limit has not been reached. The default time limit is 300 seconds, but System Admins can change it.
+
+Edit a message by clicking the **[...]** icon next to any message that you’ve sent, then click **Edit**. When you're done editing, press **ENTER** to save the changes. Message edits do not trigger new @mention notifications, desktop notifications, or notification sounds.
+
+Deleting a message
+------------------
+
+The ability to delete a message after it is sent depends on which edition you are using.
+
+Team Edition
```
Comment (esethna) [inline source/help/messaging/sending-messages.rst]: Same comment as above on TE/EE in end user docs

File: source/help/messaging/sending-messages.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
+.. _sending-messages:
+
+Messaging Basics
```
Comment (esethna) [inline source/help/messaging/sending-messages.rst]: I think we already have a doc called "Message Basics".

File: source/help/messaging/sending-messages.rst [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+================
+
+After a message has been sent, you or others can reply to it and link to it. Depending on which edition of Mattermost you have and how it's configured by the System Admin, you can also edit and delete your own messages.
+
+Message Types
+-------------
+
+Mattermost has two types of messages: posts and replies.
+
+Posts
```
Comment (esethna) [inline source/help/messaging/sending-messages.rst]: Is there a more end user friendly terminology than posts? Perhaps "messages" and "replies?" Or do we even have to make this distinction in the docs? Perhaps providing a screenshot and putting some of this info below the "Replies" heading would be suffient here.

File: source/help/messaging/sending-messages.rst [hunk 5]
```diff
@@ -0,0 +1,60 @@
+.. _sending-messages:
+
+Messaging Basics
+================
+
+After a message has been sent, you or others can reply to it and link to it. Depending on which edition of Mattermost you have and how it's configured by the System Admin, you can also edit and delete your own messages.
```
Comment (esethna) [inline source/help/messaging/sending-messages.rst]: Long winded sentence here. Is this intro necessary?

File: source/help/messaging/sending-messages.rst (hunk 3 above)
Comment (JeffSchering) [inline source/help/messaging/sending-messages.rst]: Oops. Should be Sending Messages.

File: source/help/messaging/sending-messages.rst (hunk 5 above)
Comment (JeffSchering) [inline source/help/messaging/sending-messages.rst]: Yes,

File: source/help/messaging/sending-messages.rst (hunk 4 above)
Comment (JeffSchering) [inline source/help/messaging/sending-messages.rst]: This is from the original doc. The terminology around posts, messages, and replies, should be examined at some point, but it's outside the scope of this PR.

File: source/help/messaging/sending-messages.rst (hunk 1 above)
Comment (JeffSchering) [inline source/help/messaging/sending-messages.rst]: Ok.

File: source/help/messaging/sending-messages.rst (hunk 2 above)
Comment (JeffSchering) [inline source/help/messaging/sending-messages.rst]: Ok.

**Comment by JeffSchering** [comment]
@esethna While I appreciate and understand the need for an overhaul of this doc, it is only one doc of many that need it and will be part of the rework being done for the April release. The intent of this PR is to only update for new functionality in 3.7. Reworking of this doc should be done in the context of a review and update of the entire end-user doc set, and this particular one could see a great deal of changes made to it then, so mostly it's a waste of time to do more than simply add the changes for 3.7.

Also, a couple of issues: 

 - The intro rewrite is in passive voice.
 - The menu order is incorrect:  it's Reply, Permalink, Delete, Edit.
image

Maybe you can revert the last commit?

**Comment by esethna** [comment]
Thanks Jeff, I fixed the intro, please re-review. 

I agree that an overhaul needs to be done for all end-user docs. In my mind, if we see areas for improving the readability and understanding of the docs, and the mana cost to make the change is tiny, it's not a waste of time to make that change.

**Pull request merged**